numpy/lib/npyio.py

```diff
@@ -993,8 +992,7 @@ def read_data(chunk_size):
         X = []
         line_iter = itertools.chain([first_line], fh)
         line_iter = itertools.islice(line_iter, max_rows)
-        for i, line in enumerate(line_iter):
-            vals = split_line(line)
+        for i, vals in enumerate(map(split_line, map(decode, line_iter))):
```

Comment:
While not required by variable scoping, perhaps it would be more readabl eif `decode` were passed in as an argument to `read_data`. Or would that slow things down?

Reply:
Actually, I decided to just move some of my not-yet PR'ed further improvements into this PR (second commit): I now construct an iterator (lineno_words_iter) that includes more preprocessing steps outside, and pass it to read_data.  There's no speed change associated with that specific change, but it will simplify yet another later improvement.